### DISCUSSION TITLE
pre-commit: PR still get created with minimumReleaseAge="3 days" & internalChecksFilter="strict" & scheduled

### DISCUSSION THREAD

**Discussion by trim21** [renovatebot/renovate#32829] [Request Help] [answered]
### How are you running Renovate?

A Mend.io-hosted app

### Please tell us more about your question or problem

I have these option with pre-commit manager enable:

```json
{
  "minimumReleaseAge": "3 days",
  "internalChecksFilter": "strict",
  "schedule": ["on the 1st day of the month"],
  "lockFileMaintenance": {
    "schedule": ["every 3 months on the first day of the month"]
  },
  "packageRules": [
    {
      "matchUpdateTypes": ["pin"],
      "schedule": ["at any time"]
    }
  ]
}
```

and the docs saye bot won't create a new branch or PR after new version is released after 3days

https://docs.renovatebot.com/configuration-options/#suppress-branchpr-creation-for-x-days

But I find it didn't work as expected, a PR is still create and a checks is pending

for example, https://github.com/trim21/sslog/pull/12

at the time of writing, this version is released only 2 days ago:
![image](https://github.com/user-attachments/assets/8c39a29b-efa8-422b-bb8d-9f88ca64d69f)

![image](https://github.com/user-attachments/assets/bfb1609a-8460-4007-b4b6-c6e054e76e4e)
 

### Logs (if relevant)

<details><summary>Logs</summary>

```

DEBUG: Matches schedule on the 1st day of the month (branch="renovate/astral-sh-ruff-pre-commit-0.x")
DEBUG: Update has not passed minimum release age (branch="renovate/astral-sh-ruff-pre-commit-0.x")
{
  "depName": "astral-sh/ruff-pre-commit"
  "timeElapsed": 181405654
  "minimumReleaseAge": "3 days"
}
```

full log: https://gist.github.com/trim21/05d89d80e61f7b615ce9e26ff36fb154

</details>

**Comment by RahulGautamSingh** [comment, marked as answer]
If all releases are pending then the bot picks the latest release and tries to create a branch with it (if further conditions are met), even with `internalChecksFilter=true`. 

You should set `prCreation=not-pending` (this is one of the "further conditions"), which will prevent any PR creation for updates with pending checks.

**Reply by trim21** [reply]
thanks. will `prCreation=not-pending` create branch?

**Reply by RahulGautamSingh** [reply]
No
